src/Skoruba.IdentityServer4.Admin.BusinessLogic/Helpers/EnumerableExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Skoruba.IdentityServer4.Admin.BusinessLogic.Helpers
+{
+    public static class EnumerableExtensions
+    {
+        public static IEnumerable<T> WhereIf<T>(this IEnumerable<T> query, bool condition, Func<T, bool> predicate)
+        {
+            return condition
+                ? query.Where(predicate)
+                : query;
+        }
+
+        public static IEnumerable<T> TakeIf<T, TKey>(this IEnumerable<T> query, Func<T, TKey> orderBy, bool condition, int limit, bool orderByDescending = true)
```

Comment:
Could you replace `condition` and `limit` with `int?` instead? That way you have just one argument and the coupled arguments enforce correct invariants ("no value" or "some value" instead of current  yes/no value and the actual value which can encode invalid invariant where it says no limit but the number is non-zero).